test/unit/puppetlabs/services/jruby/jruby_puppet_core_test.clj

```diff
@@ -19,3 +64,55 @@
     (is (= 4 (jruby-core/default-pool-size 16)))
     (is (= 4 (jruby-core/default-pool-size 32)))
     (is (= 4 (jruby-core/default-pool-size 64)))))
+
+(deftest ^:integration cli-run!-test
```

Comment:
Questions aside about whether or not the remapping of stdin and stdout is a "scary" thing to do in the tests, I think these tests are pretty cool / useful.  Good to know we can do testing of this kind in Clojure rather than having to just resort to using Beaker.